src/ngraph/runtime/manager.hpp

```diff
@@ -36,13 +37,17 @@ namespace ngraph
         /// a backed for execution and allocation.
         class Manager
         {
+            friend class runtime::Backend;
+
         public:
             virtual ~Manager() {}
             /// @brief Allocate a backend for this transformer.
             ///
             /// Specific transformers may provide addtional methods for allocating customized backends.
             virtual std::shared_ptr<Backend> allocate_backend() = 0;
```

Comment:
Please describe the return value i.e., if allocate fails what is the return (possibly nullptr?)
Also, how does the caller know the reason for the failure. Can this function either a failure object or an actual value - may be a good idea to adopt something similar here?
(This is applicable to error reporting in general - so may be we could address it outside of this PR)